pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -368,11 +368,44 @@ fi
 
 # Verify that user is not trying to use attribute building and flake
 # at the same time
-if [[ -z $buildingAttribute && -n $flake ]]; then
+if [[ -n $buildingFile && -n $flake ]]; then
     log "error: '--flake' cannot be used with '--file' or '--attr'"
     exit 1
 fi
 
+# Finds a specific file in a directory or its parents
+findInParents() {
+    local dir=$1
+    local filename=$2
+    while [[ ! -f "$dir/$filename" ]] && [[ "$dir" != / ]]; do
+        dir=$(dirname "$dir")
+    done
+    if [[ -f "$dir/$filename" ]]; then
+        echo "$dir/$filename"
+    else
+        return 1
+    fi
+}
+
+# Default build file if none is specified in the arguments
+if [[ -z $buildFile ]]; then
+    # From NIX_PATH
+    if ! buildFile=$(runCmd nix-instantiate --find-file nixos-system 2>/dev/null); then
+        # From system.nix up from the current directory
+        if ! buildFile=$(findInParents "$PWD" system.nix); then
+            # Hardcoded to /etc/nixos/system.nix
+            if [[ -f "/etc/nixos/system.nix" ]]; then
+                buildFile=/etc/nixos/system.nix
+            fi
```

Comment:
I don't think this path should be hardcoded in nixos-rebuild. Setup with channels doesn't have /etc/nixos/configuration.nix hardcoded and uses nixos-config in `NIX_PATH` instead to find its configuration there. If the /etc/nixos/system.nix must be hardcoded, it should be in NixOS modules instead, similar to channel setup.

Reply:
flake path `/etc/nixos/flake.nix` is hardcoded in `nixos-rebuild`, I don't think the issue that I brough up has any significance anymore